src/dune_rules/module_trie.mli

```diff
@@ -0,0 +1,50 @@
+type 'a t = 'a node Module_name.Map.t
```

Comment:
I'd suggest adding a comment explaining what this module is, for new contributors coming across it. My attempt:

```suggestion
(** This module defines a prefix tree that stores a map of module names at every
    non-leaf node. Most notably, it's used to implement `(include_subdirs
    qualified)` where the directory name qualifies the namespace for its
    descendant modules in the file system. *)

type 'a t = 'a node Module_name.Map.t
```

Reply:
@rgrinberg You missed the full stop at the end of the comment.